underscore.js

```diff
@@ -779,13 +779,21 @@
 
   // Memoize an expensive function by storing its results.
   _.memoize = function(func, hasher) {
+    function getCache() {
+      if (Object.create) {
+        return Object.create(null);
+      }
+      return {};
+    }
+
     var memoize = function(key) {
       var cache = memoize.cache;
       var address = '' + (hasher ? hasher.apply(this, arguments) : key);
-      if (!_.has(cache, address)) cache[address] = func.apply(this, arguments);
+      if (!(_.has(cache, address))) cache[address] = func.apply(this, arguments);
       return cache[address];
```

Comment:
If you aren't don't close the PR. I'll walk you through some things to improve it.